### PR TITLE
Added support for environment variable references

### PR DESCRIPTION
**Pull request by drobinson**
### Issue

This PR implements  #20 .

### Proposed changes

Here's what I was talking about.  If this looks like something you'll want to include, I can write up docs in README.  One thing I still need to resolve, is that it looks like encrypted fields do not appear to be encrypted by this tool (requiring the already encrypted value to be set in yaml).  I wonder how best to resolve this, since users will likely already be using it in this manner.  Maybe another cli option for enableEncrypt or something...

### CODE REVIEW

File: Model/Processor/ImportProcessor.php [hunk 1]
```diff
@@ -159,4 +171,30 @@ public function transformConfigToScopeConfig($path, array $config)
...
+     * @param string $value
+     * @return string|false the original string, or the resolved
+     * string if it's a reference or false on error
+     */
+    public function resolveValue($value)
+    {
+        $value = preg_replace_callback(
+            '/\{env:([^:\}\{]+?)\}/',
+            function ($matches) {
+                $value = getenv($matches[1]);
```
Comment (ktomk) [inline Model/Processor/ImportProcessor.php:190]: An environment variables resolver should be injected in the constructor (perhaps defer).

File: Model/Processor/ImportProcessor.php [hunk 2]
```diff
@@ -159,4 +171,30 @@ public function transformConfigToScopeConfig($path, array $config)
...
+                if ($value === false) {
+                    throw new \UnexpectedValueException('Environment variable does not exist');
```
Comment (ktomk) [inline Model/Processor/ImportProcessor.php:192]: The name of the missing environment variable should be given. Also variable substitution normally supports default values, which is normally pretty handy for distributed setups.

File: Model/Processor/ImportProcessor.php [hunk 3]
```diff
@@ -159,4 +171,30 @@ public function transformConfigToScopeConfig($path, array $config)
...
+     * @throws \UnexpectedValueException
+     *
+     * @param string $value
+     * @return string|false the original string, or the resolved
+     * string if it's a reference or false on error
+     */
+    public function resolveValue($value)
+    {
+        $value = preg_replace_callback(
+            '/\{env:([^:\}\{]+?)\}/',
```
Comment (ktomk) [inline Model/Processor/ImportProcessor.php:188]: I could not find any reference, but perhaps this string is always UTF-8 so the pattern should reflect that. Depends on what the YAML Symfony component does return (which is what I could not locate a normative reference for the character encoding, YAML itself supports more, but the Symfony component only a subset of it.

File: Model/Processor/ImportProcessor.php [hunk 4]
```diff
@@ -149,6 +149,18 @@ public function transformConfigToScopeConfig($path, array $config)
                     continue;
                 }
 
+                try {
+                    $value = $this->resolveValue($value);
+                } catch (\UnexpectedValueException $e) {
+                    $errorMsg = sprintf(
+                        '<error>' . $e->getMessage() . ' (%s => %s)</error>',
```
Comment (ktomk) [inline Model/Processor/ImportProcessor.php:156]: As you're using sprintf, the message can be a place-holder, too.

File: Model/Processor/ImportProcessor.php [hunk 5]
```diff
@@ -159,4 +171,30 @@ public function transformConfigToScopeConfig($path, array $config)
...
+    public function resolveValue($value)
+    {
+        $value = preg_replace_callback(
+            '/\{env:([^:\}\{]+?)\}/',
+            function ($matches) {
+                $value = getenv($matches[1]);
+                if ($value === false) {
+                    throw new \UnexpectedValueException('Environment variable does not exist');
+                }
+                return getenv($matches[1]);
```
Comment (ktomk) [inline Model/Processor/ImportProcessor.php:194]: `$value`

**Review by ktomk — COMMENTED** [review]
Just some quick comments on the changes, no concrete changes requested right now.

**Comment by ktomk** [comment]
Regarding the encryption problem: Aren't that specific to certain paths? If so, this could be possible with some decoration.

For the overall environment part, please see:

> New in version 3.2: env() parameters were introduced in Symfony 3.2.

https://symfony.com/doc/current/configuration/external_parameters.html

**Comment by peterjaap** [comment]
@therouv I'd like to pick this up since we plan on starting to use this extension but we need environment variables support.

**Pull request closed without merging**